### PR TITLE
Feature/service pagination

### PR DESCRIPTION
**Pull request by mrSelieznov**
#129 Task

- Updated API methods to use pagination
- Added method to change response without api's pages pagination to custom
- Updated manual service

### CODE REVIEW

File: src/service/api.tsx [hunk 1]
```diff
@@ -107,26 +107,30 @@ const api = {
     return obj.data;
   },
 
-  async getPopularQueryList() {
-    const obj = await axios.get(`movie/popular?${API_KEY}&query`);
+  async getPopularQueryList(page: number = 1) {
+    const obj = await axios.get(`movie/popular?${API_KEY}&query&page=${page}`);
```
Comment (oleksandrstarov) [inline src/service/api.tsx:111]: &query& ???

File: src/service/api.tsx [hunk 2]
```diff
@@ -107,26 +107,30 @@ const api = {
...
   },
 
-  changeImgLinks(url: string, size: string) {
+  changeImgLinks(url: string, size: string = 'w500') {
     return `${API_IMG_URL}${size}${url}`;
-  }
-};
+  },
+
+  changeListByPagination(arr: Array<{}>, page: number = 1) {
```
Comment (oleksandrstarov) [inline src/service/api.tsx:129]: where is return type? Array<{}> - is wrong... here you might use generics - https://www.typescriptlang.org/docs/handbook/2/generics.html
`changeListByPagination(arr: Array<T>, page: number = 1): Array<T> {`

File: src/service/api.tsx [hunk 3]
```diff
@@ -107,26 +107,30 @@ const api = {
...
+    return arr.length < 20
+      ? arr.slice(0, 20)
```
Comment (oleksandrstarov) [inline src/service/api.tsx:131]: maybe just arr? why slice?

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

File: src/service/api.tsx (hunk 3 above)
Comment (mrSelieznov) [reply src/service/api.tsx:131]: oh yes you are right

File: src/service/api.tsx [hunk 4]
```diff
@@ -117,26 +117,30 @@ const api = {
...
     return `${API_IMG_URL}${size}${url}`;
-  }
-};
+  },
+
+  changeListByPagination(arr: {}[], page: number = 1): {}[] {
+    return arr.length < 6
+      ? arr
+      : arr.slice(6 * (page - 1), 6 * page);
+  },
```
Comment (oleksandrstarov) [inline src/service/api.tsx:143]: Already left comment for this code
add types 
Array<T>
{}[] - is NOT a valid type

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

File: src/service/api.tsx (hunk 2 above)
Comment (oleksandrstarov) [reply src/service/api.tsx]: have you seen this comment @mrSelieznov it is as easy as ctrl+c > ctrl+v as solution already here

File: src/service/api.tsx [hunk 5]
```diff
@@ -84,7 +83,7 @@ const api = {
 
   isIdInHistory(id: number) {
     const store = api.getHistoryIdList();
-     return !store.includes(id);
+    return !store.includes(id);
```
Comment (oleksandrstarov) [inline src/service/api.tsx:86]: wrong condition, if Id in history it should include

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

**Review by oleksandrstarov — APPROVED** [review]

**Pull request merged**